presto-docs/src/main/sphinx/security/built-in-system-access-control.rst

```diff
@@ -133,8 +147,42 @@ and Kerberos authentication:
           "allow": true
         }
       ],
-      "user_patterns": [
-        "(.*)",
-        "([a-zA-Z]+)/?.*@.*"
+      "principals": [
+        {
+          "principal": "(.*)",
+          "user_extraction": "$1",
+          "allow": true
+        },
+        {
+          "principal": "([^/]+)/?.*@.*",
+          "user_extraction": "$1",
+          "allow": true
+        }
+      ]
+    }
+
+If you want to allow users to use the extractly same name as their Kerberos principal
+name, and allow ``alice`` and ``bob`` to use a group principal named as
+``group@example.net``, you can use the following rules.
+
+.. code-block:: json
+
+    {
+      "catalogs": [
+        {
+          "allow": true
+        }
+      ],
+      "principals": [
+        {
+          "principal": "([^/]+)/?.*@example.net",
+          "user_extraction": "$1",
+          "allow": true
+        },
+        {
+          "principal": "group@example.net",
+          "user": "alice|bob",
```

Comment:
since this is json, why not make this array of string?  (just an idea)

Reply:
I would love the flexibility of regex. If we want to go something like `presto-user-.*` in future. We won't have to add a new type of criterion 😄